architecture/understanding-development.adoc

```diff
@@ -1,45 +1,396 @@
-[id="understanding-openshift-development"]
+[id="understanding-development"]
```

Comment:
@bmcelvee, will you PTAL at the links in this file and let me know if you know of any better links within the collection?